staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_discovery.go

```diff
@@ -228,8 +229,48 @@ func (dm *discoveryManager) fetchFreshDiscoveryForService(gv metav1.GroupVersion
 
 		dm.setCacheEntryForService(info.service, cached)
 		return &cached, nil
-	case http.StatusNotAcceptable:
-		// Discovery Document is not being served at all.
+	} else if writer.respCode == http.StatusOK && discovery.ContentTypeIsGVK(writer.Header().Get("Content-Type"), schema.GroupVersionKind{
+		Group:   "apidiscovery.k8s.io",
+		Version: "v2beta1",
+		Kind:    "APIGroupDiscoveryList",
+	}) {
```

Comment:
pull the schema.GroupVersionKind{...} bit into a private package var for readability